src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
+        // Verify size of list
+        ReturnErrorOnFailure(aVlist.ComputeSize(&pairTotal));
+        VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        return CHIP_NO_ERROR;
+    }
+
     /// @brief From command AddScene, allows handler to filter through clusters in command to serialize only the supported ones.
     /// @param endpoint[in] Endpoint ID
     /// @param extensionFieldSet[in] ExtensionFieldSets provided by the AddScene Command, pre initialized
-    /// @param serialisedBytes[out] Buffer to fill from the ExtensionFieldSet in command
-    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value otherwise
+    /// @param serializedBytes[out] Buffer to fill from the ExtensionFieldSet in command
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR_INVALID_ARGUMENT if the cluster is not supported, CHIP_ERROR value
+    /// otherwise
     virtual CHIP_ERROR SerializeAdd(EndpointId endpoint,
                                     const app::Clusters::Scenes::Structs::ExtensionFieldSet::DecodableType & extensionFieldSet,
                                     MutableByteSpan & serializedBytes) override
     {
-        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair;
-        TLV::TLVWriter writer;
-        TLV::TLVType outer;
-        size_t pairTotal  = 0;
-        uint8_t pairCount = 0;
+        app::Clusters::Scenes::Structs::AttributeValuePair::Type aVPairs[kMaxAvPair];
 
+        size_t pairTotal = 0;
         // Verify size of list
         ReturnErrorOnFailure(extensionFieldSet.attributeValueList.ComputeSize(&pairTotal));
-        VerifyOrReturnError(pairTotal <= ArraySize(mAVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
+        VerifyOrReturnError(pairTotal <= ArraySize(aVPairs), CHIP_ERROR_BUFFER_TOO_SMALL);
 
+        uint8_t pairCount  = 0;
         auto pair_iterator = extensionFieldSet.attributeValueList.begin();
         while (pair_iterator.Next())
         {
-            aVPair                             = pair_iterator.GetValue();
-            mAVPairs[pairCount].attributeID    = aVPair.attributeID;
-            mAVPairs[pairCount].attributeValue = aVPair.attributeValue;
+            app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType aVPair = pair_iterator.GetValue();
+            aVPairs[pairCount].attributeID                                           = aVPair.attributeID;
+            aVPairs[pairCount].attributeValue                                        = aVPair.attributeValue;
             pairCount++;
         }
         ReturnErrorOnFailure(pair_iterator.GetStatus());
-
-        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList;
-        attributeValueList = mAVPairs;
+        app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList(aVPairs);
```

Comment:
A lot of this code could benefit from some judicious "using" in this class scope... ;) Followup for that?

Reply:
Issue logged [here](https://github.com/project-chip/connectedhomeip/issues/27111)